packages/data/addon/adapters/bard-facts-v2.ts

```diff
@@ -0,0 +1,408 @@
+    const dimensions = array(request.columns)
+      .filterBy('type', 'dimension')
+      .map(dim => formatDimensionFieldName(dim.field, dim.parameters, false));
+
+    return dimensions.length
+      ? `/${array(dimensions)
+          .uniq()
+          .join('/')}`
+      : '';
+  }
+
+  /**
+   * Builds a dateTime param string for a request
+   *
+   * @method _buildDateTimeParam
+   * @private
+   * @param {Object} request
+   * @return {String} dateTime param value
+   */
+  _buildDateTimeParam(request: RequestV2): string {
+    const dateTimeFilters = request.filters.filter(fil => fil.field === 'dateTime');
+
+    return dateTimeFilters.map(filter => `${filter.values[0]}/${filter.values[1]}`).join(',');
+  }
+
+  /**
+   * Builds a metrics param string for a request
+   *
+   * @method _buildMetricsParam
+   * @private
+   * @param {Object} request
+   * @return {String} metrics param value
+   */
+  _buildMetricsParam(request: RequestV2): string {
+    const metrics = request.columns.filter(col => col.type === 'metric');
+    const metricIds = metrics.map(metric =>
+      canonicalizeMetric({ metric: metric.field, parameters: metric.parameters })
+    );
+
+    return array(metricIds)
+      .uniq()
+      .join(',');
+  }
+
+  /**
+   * Builds a filters param string for a request
+   *
+   * @method _buildFiltersParam
+   * @private
+   * @param {Object} request
+   * @return {String} filters param value
+   */
+  _buildFiltersParam(request: RequestV2) {
+    const filters = request.filters.filter((fil: Filter) => fil.type === 'dimension');
+
+    if (filters?.length) {
+      return serializeFilters(filters);
+    }
+    return undefined;
+  }
+
+  /**
+   * Builds a sort param string for a request
+   *
+   * @method _buildSortParam
+   * @private
+   * @param {Object} request
+   * @param {function} aliasFunction function that returns metrics from aliases
+   * @return {String} sort param value
+   */
+  _buildSortParam(request: RequestV2, aliasFunction: AliasFn = a => a) {
+    const { sort } = request;
+
+    if (sort.length) {
+      return sort
+        .map(sortField => {
+          const field = aliasFunction(sortField.field);
+          const direction = sortField.direction || 'desc';
+
+          assert(
+            `'${direction}' must be a valid sort direction (${SORT_DIRECTIONS.join()})`,
+            SORT_DIRECTIONS.includes(direction)
+          );
+
+          return `${field}|${direction}`;
+        })
+        .join(',');
+    }
+    return undefined;
+  }
+
+  /**
+   * Builds a having param string for a request
+   *
+   * @method _buildHavingParam
+   * @private
+   * @param {Object} request
+   * @param {function} aliasFunction function that returns metrics from aliases
+   * @return {String} having param value
+   */
+  _buildHavingParam(request: RequestV2, aliasFunction: AliasFn = a => a) {
+    const having = request.filters.filter(fil => fil.type === 'metric');
```

Comment:
Filters don't have type field. have to use a different method for distinction. Probably can use columnMetaType

Reply:
Filters are going to have a type. I discussed it with Jon and opened a PR against the request v2 RFC. We should probably add type to Sort as well.

Reply:
Yeah we should add it to sorts, (we should add it to the base fragment )